src/sssom/util.py

```diff
@@ -246,7 +246,7 @@ def clean_prefix_map(self, strict: bool = True) -> None:
         if self.metadata:
             prefixes_in_table.update(get_prefixes_used_in_metadata(self.metadata))
 
-        missing_prefixes = prefixes_in_table - self.converter.get_prefixes()
+        missing_prefixes = prefixes_in_table - self.converter.get_prefixes(include_synonyms=True)
```

Comment:
Why this change?

Reply:
This overwrote the changes Joe made with an equivalent upstream improvement to the `curies` package. I still don't think we should add this here, but this is for posterity. Ultimately I hope this PR will get rejected in favor of solving the data problems upstream.

Reply:
Big thanks @cthoyt for contributing your updates to the PR! I thought I was going to have to do that.

However the changes in this function `clean_prefix_map()` don't seem like they'd involve an upstream data fix. This is just a bugfix. Sorry, as mentioned there are 2 things in this PR and maybe it would've been best to fix: (1) for handling bijective maps and converting plain prefix maps to EPMs, and (2) fixing `clean_prefix_map()` so that it detects prefix synonyms. Otherwise it throws an error because it thinks prefixes are missing.

Reply:
@joeflack4 I'll revert this change and leave it for you to open a new PR that includes the update and a unit test that demonstrates the bug

Reply:
Hmm, I think one or both of us is confused. You don't need to revert anything. Your change to move this logic from `sssom-py`'s `clean_prefix_map()` to `curies` is good. I wanted this logic in `curies` originally, and I think the `get_prefixes(include_synonyms=True)` solution is good.

I could make a test but it's probably redundant if you've already tested the `include_synonyms` param in `curies`.

I think this one line update adding `include_synonyms=True` is probably the only change needed in `sssom-py` now to address my issues. I'm likely going to close this PR and open a new one with just those changes (as well as the pyproject.tom / poetry.lock updates).

Reply:
This PR is essentially a 1-liner now along with updated `pyproject.toml` and `poetry.lock` for `curies` upgrade.

Reply:
This PR is essentially a 1-liner now along with updated `pyproject.toml` and `poetry.lock` for `curies` upgrade.